scripts/sync-gomod.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+LIB_DIR="$DIR/../.bootstrap/shell/lib"
+
+# shellcheck source=../.bootstrap/shell/lib/logging.sh
+source "$LIB_DIR/logging.sh"
+
+run_sed() {
+  case "$OSTYPE" in
+  darwin*) # macOS
+    sed -i '' -e "$@"
+    ;;
+  linux*)
+    sed -i -e "$@"
+    ;;
+  esac
+}
```

Comment:
Admittedly I did not test this on macOS, but it's very similar to this function:

https://github.com/getoutreach/devbase/blob/1ff9c81010a7e7771d29971daf45dde17f9eaa8c/shell/catalog-sync.sh#L14-L28

Reply:
Didn't run the whole thing but verified the mac command